platform-hub-web/src/app/app.routes.spec.js

```diff
@@ -0,0 +1,373 @@
+/* eslint angular/log: 0, padded-blocks: 0 */
+
+import angular from 'angular';
+import 'angular-mocks';
+import lodash from 'lodash';
+import sinon from 'sinon';
+import chai from 'chai';
+import 'chai/register-should';
+import sinonChai from 'sinon-chai';
+chai.use(sinonChai);
+
+describe('routes', () => {
+  const PARAM_REGEX = /(:\w+)/g;
+
+  const ALLOWED_DATA_CONFIG_FIELDS = [
+    'authenticate',
+    'rolePermitted',
+    'featureFlag'
+  ];
+
+  let sandbox = null;
+
+  let $q = null;
+  let $httpBackend = null;
+  let $state = null;
+  let $location = null;
+  let $rootScope = null;
+  let authService = null;
+  let loginDialogService = null;
+  let FeatureFlags = null;
+  let roleCheckerService = null;
+  let Me = null;
+
+  beforeEach(() => {
+    const moduleName = 'app.routes.spec';
+    angular.module(moduleName, ['app']);
+    angular.mock.module(moduleName);
+  });
+
+  beforeEach(angular.mock.inject((_$q_, _$httpBackend_, _$state_, _$location_, _$rootScope_, _authService_, _loginDialogService_, _FeatureFlags_, _roleCheckerService_, _Me_) => {
+    sandbox = sinon.sandbox.create();
+
+    $q = _$q_;
+    $httpBackend = _$httpBackend_;
+    $state = _$state_;
+    $location = _$location_;
+    $rootScope = _$rootScope_;
+    authService = _authService_;
+    loginDialogService = _loginDialogService_;
+    FeatureFlags = _FeatureFlags_;
+    roleCheckerService = _roleCheckerService_;
+    Me = _Me_;
+
+    $httpBackend
+      .whenGET('/api/app_settings')
+      .respond('{}');
+  }));
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('route config', () => {
+    it(`should only allow certain fields in the 'data' object`, () => {
+      function stubs() { }
+
+      function checks(state) {
+        lodash.keys(state.data).forEach(k => {
+          expect(ALLOWED_DATA_CONFIG_FIELDS).toContain(k, `in 'data' config for route: ${state.name}`);
+        });
+      }
+
+      eachNavigableState(stubs, checks);
+    });
+  });
+
+  describe(`for not logged in users who don't log in`, () => {
```

Comment:
Unfortunately, Jasmine is inferior to RSpec. No `context` blocks (amongst other things).